### PR TITLE
Prefer future<Unit> to future<bool>

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
https://github.com/facebook/folly/blob/master/folly/Try.h
From the above link:

```
 * Try<T> is a wrapper that contains either an instance of T, an exception, or
 * nothing. Exceptions are stored as exception_wrappers so that the user can
 * minimize rethrows if so desired.
 *
 *To represent success or a captured exception, use Try<Unit>.*
```

Try < Unit >  is also implicity convertible to Try < void >. This change uses folly::exception_wrapper as a lightweight way to represent and forward errors in futures that don't really have a value to return, thus the Unit return type

File: connectors/cassandra/include/CassandraInserter.hpp [hunk 1]
```diff
@@ -37,7 +38,7 @@ class CassandraInserter {
 private:
   folly::Optional<std::string> setSchemaMetadata();
   folly::Optional<std::vector<std::string>> queryTableMetadata();
-  folly::Future<bool> registerCallback(CassFuture *ft);
+  folly::Future<folly::Unit> registerCallback(CassFuture *ft);
```
Comment (emaxerrno) [inline connectors/cassandra/include/CassandraInserter.hpp:41]: you should alias this as ``folly::Future<>`

File: connectors/cassandra/include/CassandraInserter.hpp [hunk 2]
```diff
@@ -19,7 +19,8 @@ class CassandraInserter {
   folly::Future<folly::Optional<std::string>> connectSession();
   void disconnectSession();
 
-  template <typename Func> folly::Future<bool> insert(Func &&insertCallback) {
+  template <typename Func>
+  folly::Future<folly::Unit> insert(Func &&insertCallback) {
```
Comment (emaxerrno) [inline connectors/cassandra/include/CassandraInserter.hpp:23]: just return 

```

folly::Future<> 
```

File: connectors/cassandra/include/CassandraSink.hpp [hunk 3]
```diff
@@ -32,7 +32,7 @@ class CassandraSink : public bolt::Computation {
   uint64_t failedCallbacks_{0};
   const bolt::Metadata thisMetadata_;
   CassandraInserter cluster_;
-  std::vector<folly::Future<bool>> asyncInserts_;
+  std::vector<folly::Future<folly::Unit>> asyncInserts_;
```
Comment (emaxerrno) [inline connectors/cassandra/include/CassandraSink.hpp:35]: same here

File: connectors/cassandra/lib/CassandraInserter.cc [hunk 4]
```diff
@@ -94,26 +95,30 @@ CassandraInserter::queryTableMetadata() {
...
-folly::Future<bool> CassandraInserter::registerCallback(CassFuture *ft) {
+folly::Future<folly::Unit> CassandraInserter::registerCallback(CassFuture *ft) {
   auto f = wrapCassFuture(ft);
-  auto promise = std::make_unique<folly::Promise<bool>>();
+  auto promise = std::make_unique<folly::Promise<folly::Unit>>();
   auto future = promise->getFuture();
   if (cass_future_set_callback(f.release(), databaseEffectCallback,
                                (void *)promise.release()) != CASS_OK) {
-    return folly::Future<bool>(false);
+    promise->setException(folly::make_exception_wrapper<std::runtime_error>(
```
Comment (emaxerrno) [inline connectors/cassandra/lib/CassandraInserter.cc:104]: :+1:

File: connectors/cassandra/lib/CassandraInserter.cc [hunk 5]
```diff
@@ -94,26 +95,30 @@ CassandraInserter::queryTableMetadata() {
...
   }
   return future;
 }
 
 // static
 void CassandraInserter::databaseEffectCallback(CassFuture *ft, void *data) {
-  auto promise =
-      std::unique_ptr<folly::Promise<bool>>((folly::Promise<bool> *)data);
+  auto promise = std::unique_ptr<folly::Promise<folly::Unit>>(
+      (folly::Promise<folly::Unit> *)data);
```
Comment (emaxerrno) [inline connectors/cassandra/lib/CassandraInserter.cc:113]: you really need to typedef this thing.

File: connectors/cassandra/lib/CassandraSink.cc [hunk 6]
```diff
@@ -63,15 +63,13 @@ void CassandraSink::processRecord(CtxPtr ctx, bolt::FrameworkRecord &&r) {
...
         for (const auto &t : tries) {
-          if (t.hasException()) {
-            LOG(ERROR) << "Exception detected in future: "
-                       << t.exception().what();
-            failedCallbacks_ += 1;
-          } else if (t.hasValue() && t.value() == false) {
-            failedCallbacks_ += 1;
-          }
+          const auto ex =
+              t.withException<std::runtime_error>([](const std::exception &e) {
```
Comment (emaxerrno) [inline connectors/cassandra/lib/CassandraSink.cc:69]: did you run the clang formatter. this looks a bit off.

File: connectors/cassandra/lib/CassandraInserter.cc (hunk 5 above)
Comment (ghost) [inline connectors/cassandra/lib/CassandraInserter.cc:113]: lol

**Pull request merged**